sermant-plugins/sermant-router/router-common/src/main/java/io/sermant/router/common/utils/XdsRouterUtil.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2024-2024 Sermant Authors. All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package io.sermant.router.common.utils;
+
+import io.sermant.core.config.ConfigManager;
+import io.sermant.core.plugin.config.ServiceMeta;
+import io.sermant.core.service.ServiceManager;
+import io.sermant.core.service.xds.XdsCoreService;
+import io.sermant.core.service.xds.XdsServiceDiscovery;
+import io.sermant.core.service.xds.entity.ServiceInstance;
+import io.sermant.core.service.xds.entity.XdsLocality;
+import io.sermant.core.utils.NetworkUtils;
+import io.sermant.core.utils.StringUtils;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * XdsRouterUtil
+ *
+ * @author daizhenyu
+ * @since 2024-08-29
+ **/
+public class XdsRouterUtil {
+    private static XdsServiceDiscovery serviceDiscovery = ServiceManager.getService(XdsCoreService.class)
+            .getXdsServiceDiscovery();
+
+    private static XdsLocality xdsLocality;
+
+    private XdsRouterUtil() {
+    }
+
+    /**
+     * get XdsLocality of self-service
+     *
+     * @return XdsLocality
+     */
+    public static Optional<XdsLocality> getLocalityInfoOfSelfService() {
+        if (xdsLocality != null) {
+            return Optional.of(xdsLocality);
+        }
+        synchronized (XdsRouterUtil.class) {
+            if (xdsLocality != null) {
+                return Optional.of(xdsLocality);
+            }
+            Set<ServiceInstance> serviceInstances = serviceDiscovery
+                    .getServiceInstance(ConfigManager.getConfig(ServiceMeta.class).getService());
+            String podIp = NetworkUtils.getKubernetesPodIp();
+            if (StringUtils.isEmpty(podIp)) {
+                return Optional.empty();
+            }
```

Comment:
You are advised to check whether podId is empty before executing serviceDiscovery.
.getServiceInstance(ConfigManager.getConfig(ServiceMeta.class).getService())

Reply:
done